Mathlib/Algebra/GroupPower/Lemmas.lean

```diff
@@ -2,35 +2,1345 @@
+  any_goals simp [a0.le, b0.le, abs_of_nonpos, add_nonpos, add_comm]
+  have : (|a + b| = -a + b ↔ b ≤ 0) ↔ (|a + b| = |a| + |b| ↔ 0 ≤ a ∧ 0 ≤ b ∨ a ≤ 0 ∧ b ≤ 0) := by
+    simp [a0, a0.le, a0.not_le, b0, abs_of_neg, abs_of_nonneg]
+  refine' this.mp ⟨fun h => _, fun h => by simp only [le_antisymm h b0, abs_of_neg a0, add_zero]⟩
+  obtain ab | ab := le_or_lt (a + b) 0
+  · refine' le_of_eq (eq_zero_of_neg_eq _)
+    rwa [abs_of_nonpos ab, neg_add_rev, add_comm, add_right_inj] at h
+  · refine' (lt_irrefl (0 : α) _).elim
+    rw [abs_of_pos ab, add_left_inj] at h
+    rwa [eq_zero_of_neg_eq h.symm] at a0
+#align abs_add_eq_add_abs_le abs_add_eq_add_abs_le
+
+theorem abs_add_eq_add_abs_iff (a b : α) : |a + b| = |a| + |b| ↔ 0 ≤ a ∧ 0 ≤ b ∨ a ≤ 0 ∧ b ≤ 0 := by
+  obtain ab | ab := le_total a b
+  · exact abs_add_eq_add_abs_le ab
+  · rw [add_comm a, add_comm (abs _), abs_add_eq_add_abs_le ab, and_comm, @and_comm (b ≤ 0)]
+#align abs_add_eq_add_abs_iff abs_add_eq_add_abs_iff
+
+end LinearOrderedAddCommGroup
+
+-- Porting note: commented out as linter reports LHS= RHS syntactically
+-- @[simp]
+-- theorem WithBot.coe_nsmul [AddMonoid A] (a : A) (n : ℕ) : ((n • a : A) : WithBot A) = n • a :=
+--   by simp
+-- #align with_bot.coe_nsmul WithBot.coe_nsmul
+
+/- warning: nsmul_eq_mul' -> nsmul_eq_mul' is a dubious translation:
+lean 3 declaration is
+  forall {R : Type.{u1}} [_inst_1 : NonAssocSemiring.{u1} R] (a : R)
+   (n : Nat), Eq.{succ u1} R (HasSmul.smul.{0, u1} Nat R
+   (AddMonoid.SMul.{u1} R (AddMonoidWithOne.toAddMonoid.{u1} R
+   (AddCommMonoidWithOne.toAddMonoidWithOne.{u1} R
+   (NonAssocSemiring.toAddCommMonoidWithOne.{u1} R _inst_1)))) n a)
+   (HMul.hMul.{u1, u1, u1} R R R (instHMul.{u1} R (Distrib.toHasMul.{u1} R
+   (NonUnitalNonAssocSemiring.toDistrib.{u1} R
+   (NonAssocSemiring.toNonUnitalNonAssocSemiring.{u1} R _inst_1))))
+   a ((fun (a : Type) (b : Type.{u1}) [self : HasLiftT.{1, succ u1} a b] => self.0)
+   Nat R (HasLiftT.mk.{1, succ u1} Nat R (CoeTCₓ.coe.{1, succ u1} Nat R
+   (Nat.castCoe.{u1} R (AddMonoidWithOne.toNatCast.{u1} R
+   (AddCommMonoidWithOne.toAddMonoidWithOne.{u1} R
+   (NonAssocSemiring.toAddCommMonoidWithOne.{u1} R _inst_1)))))) n))
+but is expected to have type
+  forall {R : Type.{u1}} [_inst_1 : Semiring.{u1} R] (a : R) (n : Nat),
+   Eq.{succ u1} R (HSMul.hSMul.{0, u1, u1} Nat R R (instHSMul.{0, u1}
+   Nat R (AddMonoid.SMul.{u1} R (AddMonoidWithOne.toAddMonoid.{u1} R
+   (AddCommMonoidWithOne.toAddMonoidWithOne.{u1} R (NonAssocSemiring.toAddCommMonoidWithOne.{u1} R
+   (Semiring.toNonAssocSemiring.{u1} R _inst_1)))))) n a)
+   (HMul.hMul.{u1, u1, u1} R R R (instHMul.{u1} R
+   (NonUnitalNonAssocSemiring.toMul.{u1} R (NonAssocSemiring.toNonUnitalNonAssocSemiring.{u1}
+   R (Semiring.toNonAssocSemiring.{u1} R _inst_1)))) a
+   (Nat.cast.{u1} R (Semiring.toNatCast.{u1} R _inst_1) n))
+Case conversion may be inaccurate. Consider using '#align nsmul_eq_mul' nsmul_eq_mul'ₓ'. -/
```

Comment:
These warnings should be removed, and add an `ₓ` to the end of the #align, along with a comment, e.g.
`-- typeclasses do not match up exactly`.